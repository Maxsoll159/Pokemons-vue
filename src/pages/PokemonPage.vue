<template>
    <h1 class="titulo">QUE TANTO SABES DE POKEMON</h1>
    <h1  class="respuesta">Respuestas correctas: {{respuestasBuenas}}</h1>
    <h1 class="respuesta">Vidas   </h1>
         <div v-if="vidas === 3">
             <img src="@/img/vida2.png" alt="" width="60"  class="espacio">
             <img src="@/img/vida2.png" alt="" width="60" class="espacio">
             <img src="@/img/vida2.png" alt="" width="60" class="espacio">
         </div>
         <div v-if="vidas === 2"  >
             <img src="@/img/vida2.png" alt="" width="60" class="espacio">
             <img src="@/img/vida2.png" alt="" width="60" class="espacio">
         </div>
         <div v-if="vidas === 1">
             <img src="@/img/vida2.png" alt="" width="60" class="espacio">
         </div>
 
    
    <h1 v-if="!pokemon">Espere por favor...</h1>

    <div v-else>

        <h1 class="respuesta">Â¿Quien es este Pokemon?</h1>
        <!-- IMAGEN DEL POKEMON-->
        <PokemonPicture :pokemonId="pokemon.id" :showPokemon="showPokemon"/>
        <!-- OPCIONES-->
        <PokemonOptions :pokemons="pokemonArr" 
        @selection="checkAnswer"
        />
        <template v-if="showAnswer">
            <h2 class="respuesta">{{message}}</h2>
            <button
            v-if="vidas === 0"
            @click="newGame()"
            >Nuevo Juego</button>
        </template>
       

    </div>

    

</template>

<script>
import PokemonPicture from '@/components/PokemonPicture.vue'
import PokemonOptions from '@/components/PokemonOptions.vue'
import getPokemonOptions from '@/helpers/getPokemonOptions'
//console.log(getPokemonOptions())
export default {
    components: {
    PokemonPicture,
    PokemonOptions
    },
    data(){
        return{
            pokemonArr: [],
            pokemon: null,
            showPokemon: false,
            showAnswer: false,
            message: '',
            vidas: 3,
            respuestasBuenas: 0
        }
    },
    methods: {
        async mixPokemonArray(){
            this.pokemonArr = await getPokemonOptions()
            const rndInt = Math.floor(Math.random()*4)
            this.pokemon = this.pokemonArr[rndInt]
        },
        checkAnswer(pokemonId){
            this.showPokemon = true
            this.showAnswer = true
            if(this.pokemon.id == pokemonId){     

                this.message = `Felicidades Campeon el nombre es ${this.pokemon.name}`
                this.respuestasBuenas++
                setTimeout(() => {
                    this.showPokemon = false
                    this.showAnswer = false
                    this.pokemonArr = []
                    this.mixPokemonArray()

                }, 2000)
                
            }else{
                this.message = `Fallaste era,  ${this.pokemon.name}`

                this.vidas--
                if(this.vidas !== 0){
                    setTimeout(() => {
                        this.showPokemon = false
                        this.showAnswer = false
                        this.pokemonArr = []
                        this.mixPokemonArray()

                    }, 2000)
                }

            }
            
        }, 

        newGame(){
           this.showPokemon = false
           this.showAnswer = false
            this.pokemonArr = []
            this.mixPokemonArray()
            this.pokemon = null
            this.vidas = 3
        }
    },
    mounted(){
        this.mixPokemonArray()
    }
}
</script>
<style scoped>
  .titulo {
      font-family: cursive;
      color: black;
      font-size: 50px;
  }
  .espacio{
      margin-right: 10px;
  }
.respuesta{
    font-family: cursive;
    color: black;
    font-size: 35px;
}
</style>