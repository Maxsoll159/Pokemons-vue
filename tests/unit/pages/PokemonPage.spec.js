import { shallowMount, mount } from "@vue/test-utils"
import PokemonPage from '@/pages/PokemonPage'
import {pokemons} from '../mocks/pokemos.mock'
import getPokemonOptions from "@/helpers/getPokemonOptions"
let wrapper
describe('pokemonPage Component', ()=>{

beforeEach(()=>{
    wrapper = shallowMount(PokemonPage)
})

test('debe de hacer math con el snapshot', () => { 
    expect(wrapper.html()).toMatchSnapshot()
})
test('debe de llamar mixPokemonArray al montar', () => { 
    
    const mixPokemonArray = jest.spyOn(PokemonPage.methods, 'mixPokemonArray')
    const wrapper = shallowMount(PokemonPage)
    expect(mixPokemonArray).toHaveBeenCalled()

})
test('debe de hacer march con el snapshot cuando carge los pokemons', () => { 
    const wrapper = shallowMount(PokemonPage, {
        data(){
            return{
                pokemonArr: pokemons,
                pokemon: pokemons[1],
                showPokemon: false,
                showAnswer: false,
                message: '',
                vidas: 3,
                respuestasBuenas: 0
            }
        }
    })
    expect(wrapper.html()).toMatchSnapshot()
 })

 test('debe de monstrar los componentes de pokemonPicture y PokemonOptions', () => { 
    const wrapper = shallowMount(PokemonPage, {
        data(){
            return{
                pokemonArr: pokemons,
                pokemon: pokemons[1],
                showPokemon: false,
                showAnswer: false,
                message: '',
                vidas: 3,
                respuestasBuenas: 0
            }
        }
    })
    const pokemonPicture = wrapper.find('pokemon-picture-stub')
    const PokemonOptions = wrapper.find('pokemon-options-stub')
    expect(pokemonPicture.exists()).toBeTruthy()
    expect(PokemonOptions.exists()).toBeTruthy()
    expect(pokemonPicture.attributes('pokemonid')).toBe("2")
    expect(PokemonOptions.attributes('pokemons')).toBeTruthy()
    //expect(PokemonOptions.attributes('pokemons').length).toBe(4)
})
test(' pruebas con checkAnswer', async() => { 

    const wrapper = shallowMount(PokemonPage, {
        data(){
            return{
                pokemonArr: pokemons,
                pokemon: pokemons[1],
                showPokemon: false,
                showAnswer: false,
                message: '',
                vidas: 3,
                respuestasBuenas: 0
            }
        }
    })

    /*await wrapper.vm.checkAnswer(1)
    expect(wrapper.find('h2').exists()).toBeTruthy()
    expect(wrapper.vm.showPokemon).toBe(true)
    expect(wrapper.find('h2').text()).toBe(`Correcto, ${pokemons[0].name}`)*/

    await wrapper.vm.checkAnswer(10)
    expect(wrapper.vm.message).toBe(`Fallaste era,  ${pokemons[1].name}`)

 })

})
