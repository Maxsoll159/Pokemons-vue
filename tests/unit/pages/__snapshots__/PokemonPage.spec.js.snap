// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pokemonPage Component debe de hacer march con el snapshot cuando carge los pokemons 1`] = `
<h1 class="titulo">QUE TANTO SABES DE POKEMON</h1>
<h1 class="respuesta">Respuestas correctas: 0</h1>
<h1 class="respuesta">Vidas </h1>
<div><img src="" alt="" width="60" class="espacio"><img src="" alt="" width="60" class="espacio"><img src="" alt="" width="60" class="espacio"></div>
<!--v-if-->
<!--v-if-->
<div>
  <h1 class="respuesta">Â¿Quien es este Pokemon?</h1><!-- IMAGEN DEL POKEMON-->
  <pokemon-picture-stub pokemonid="2" showpokemon="false"></pokemon-picture-stub><!-- OPCIONES-->
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
  <!--v-if-->
</div>
`;

exports[`pokemonPage Component debe de hacer math con el snapshot 1`] = `
<h1 class="titulo">QUE TANTO SABES DE POKEMON</h1>
<h1 class="respuesta">Respuestas correctas: 0</h1>
<h1 class="respuesta">Vidas </h1>
<div><img src="" alt="" width="60" class="espacio"><img src="" alt="" width="60" class="espacio"><img src="" alt="" width="60" class="espacio"></div>
<!--v-if-->
<!--v-if-->
<h1>Espere por favor...</h1>
`;
